### PR TITLE
Testing daemon runner

### PR DESCRIPTION
**Pull request by andrewxhill**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,16 +1,23 @@
-version: 2.1
+version: 2
 
 defaults: &defaults
     working_directory: ~/js-threads-client
     docker:
-        - image: circleci/node:10-browsers
+        - image: textile/builder:1.13.1
```
Comment (andrewxhill) [inline .circleci/config.yml]: just happens to have most of what i needed to test this. node 10 + golang

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
             - checkout
             - restore_cache:
                 key: dependency-cache-{{ checksum "package-lock.json" }}
+            - run:
+                name: install daemon
+                background: true
+                command: |
+                    cd ~ && mkdir gosrc && cd gosrc
+                    git clone https://github.com/textileio/go-textile-threads.git && cd go-textile-threads
+                    go run daemon/main.go
```
Comment (andrewxhill) [inline .circleci/config.yml]: the above could be wrapped up in a new docker image instead for this purpose

File: .circleci/config.yml [hunk 3]
```diff
@@ -20,11 +27,11 @@ jobs:
...
-            # - run:
-            #     name: Test
-            #     command: |
-            #         mkdir -p allure-results
-            #         npm run test
+            - run:
+                name: Test
+                command: |
+                    mkdir -p allure-results
+                    npm run test:node
```
Comment (andrewxhill) [inline .circleci/config.yml]: when we create the new docker image, we'll just have to include a headless firefox install to do full test suite

File: .circleci/config.yml [hunk 4]
```diff
@@ -1,16 +1,23 @@
-version: 2.1
+version: 2
```
Comment (andrewxhill) [inline .circleci/config.yml]: circleci CLI didn't want to help me with any 2.1 stuff, so just dropped this to 2 for now

File: .circleci/config.yml (hunk 1 above)
Comment (carsonfarmer) [reply .circleci/config.yml]: Nice

File: .circleci/config.yml (hunk 2 above)
Comment (carsonfarmer) [reply .circleci/config.yml]: Good idea, create a ticket?

**Pull request merged**